spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -137,7 +138,19 @@
     }
     // Try catalog based name based resolution
     try {
-      return spark.table(tableName + "." + type);
+      if (tableName.startsWith("spark_catalog")) {
+        // Do to the design of Spark, we cannot pass multi element namespaces to the session catalog
+        // We also don't know whether the Catalog is Hive or Hadoop Based, so we will try to load it
+        // in the hive manner first, then fall back and try the location if we have completely run out of options
+        // TODO remove this when we have Spark workaround for multipart identifiers in SparkSessionCatalog
+        try {
+          return noCatalogReader.load(tableName.replaceFirst("spark_catalog\\.", "") + "." + type);
```

Comment:
One huge drawback to this is we can't actually use any Catalog Options that have been specified, so if we have options in the future than alter the way a Metadata table is read we won't have them here.

Reply:
If we could extend out into BaseSpark3Actions this would be a good place to diverge, since I think we could load the metadata table directly from the catalog, and bypass the name resolution code. But we can't do that without redoing the hierarchy or having the Rewrite Actions use reflection to determine their "loadMetadataTable" Method

Reply:
Still to figure out is how we do this, since the current method is static and it must be because the hierarchy is currently setup such that some spark actions do not defend from this class